### PR TITLE
ISPN-8323 Cache.evict(...) throws NPE if entry was already evicted and

### PR DESCRIPTION
**Pull request by wburns**
an eviction listener is present

* Use regular for loop instead of Collectors

https://issues.jboss.org/browse/ISPN-8323

### CODE REVIEW

**Comment by wburns** [comment]
This needs to be backported to 9.1.x as well after this is integrated.  Should cherrypick as is.

File: core/src/test/java/org/infinispan/eviction/impl/EvictionFunctionalTest.java [hunk 1]
```diff
@@ -91,6 +91,22 @@ public void testSimpleEvictionMaxEntries() throws Exception {
...
+      cache.put(key, value);
+
+      cache.evict(key);
+
+      assertEquals(1, evictionListener.evictedEntries.size());
+
+      // Make sure if we evict again that it doesn't increase count
+      cache.evict(key);
+
+      // TODO: this seems like a bug, but many tests rely on this - maybe change later
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/eviction/impl/EvictionFunctionalTest.java]: I'm sure if we had many tests that evicted an entry twice we'd have noticed the NPE earlier :)

Comment (wburns) [reply core/src/test/java/org/infinispan/eviction/impl/EvictionFunctionalTest.java]: Yeah but this comment is the fact that an evict causes an event even when there as nothing to evict. :(

**Comment by tristantarrant** [comment]
Merged to master and 9.1.x

**Pull request merged**